mcux/scripts/kinetis_signal2dts.py

```diff
@@ -0,0 +1,156 @@
+import re
+import os
+import sys
+
+signal_xml_fn = "signal_configuration.xml"
+
+path = "."
+
+def parse_signal_xml(signal_fn):
+    signal_tree = ET.parse(signal_fn)
+    signal_root = signal_tree.getroot()
+
+    pins = []
+
+    partNum = signal_root.find("./part_information/part_number").get('id')
+
+    print (f"################## {partNum} ###########")
+
+    periphs_node = signal_root.find("peripherals")
+    periphs = []
+    for p in periphs_node:
+        id = p.attrib.get("id")
+        name = p.attrib.get("name")
+
+        if id != name:
+            print("WARN: id and name don't match")
+
+        periphs.append(id)
+
+    pins_node = signal_root.find("pins")
+    for p in pins_node:
+        name = p.attrib.get("name")
+
+        # Kinetis family uses PT[A-Z][0-31]
+        if "PT" in name:
+#            print(name)
+
+            functions = []
+
+            num_conns = 0
+
+            for conns in p.findall("connections"):
+                num_conns += 1
+#                print(f"\t{num_conns} - {conns.tag} {conns.attrib}")
+                func = conns.attrib.get("package_function")
+
+                conns_tree = ET.ElementTree(conns)
+                periph = None
+
+                num_periph_sig = 0
+
+                for c in conns_tree.iter('peripheral_signal_ref'):
+                    num_periph_sig += 1
+#                    print(f"\t\t{num_periph_sig} - {c.tag} {c.attrib}")
+                    signal = c.attrib.get("signal")
+                    periph = c.attrib.get("peripheral")
+                    ch = c.attrib.get("channel")
+
+                if num_periph_sig > 1:
+                    print("ERROR - %s" % num_periph_sig)
+
+                # handle unrouted_pin_function_ref case (disabled pin)
+                if periph is None:
+                    continue
+
+                pin = None
+
+                for c in conns_tree.iter('assign'):
+#                    print(f"\t\t\t[ASSIGN] {c.tag} {c.attrib}")
+                    reg = c.attrib.get("register")
+                    reg_match = re.search(r"PORT([A-Z])_PCR(\d+)", reg)
+                    if reg_match:
+                        port = reg_match.group(1).lower()
+                        if len(reg_match.groups()) == 2:
+                            pin = reg_match.group(2).lower()
+                        else:
+                            pin = None
+
+#                    print(c.attrib)
+
+                    bf = c.attrib.get("bit_field")
+                    if bf == "MUX" or bf == "FSEL" or bf == "FUNC" or "MUX_MODE":
+                        mask = c.attrib.get("bit_field_mask")
+                        val = c.attrib.get("bit_field_value")
+
+#                    print(f"  {periph} {signal} {port} {pin} {ch} {reg} {val}")
+                    pins.append((periph, signal, port, pin, ch, reg, val))
+
+    return (partNum, pins)
+
+_PIN_PERIPH = 0
+_PIN_SIGNAL = 1
+_PIN_PORT = 2
+_PIN_PIN = 3
+_PIN_CH = 4
+_PIN_REG = 5
+_PIN_VAL = 6
```

Comment:
Enum could be used, better move to the top
```suggestion
class PIN(Enum):
    PERIPH = 0
    SIGNAL = 1
    PORT = 2
    PIN = 3
    CH = 4
    REG = 5
    VAL = 6
```